### ISSUE TITLE
Yaml serialization with JsonSubTypes and JsonTypeInfo

### ISSUE THREAD

**Issue by shawkins** [fabric8io/kubernetes-client#2989]
If you use the Serialization util to create yaml from an object that uses JsonSubTypes and JsonTypeInfo, the result will not be valid yaml:

```
package org.acme;

import com.fasterxml.jackson.annotation.JsonSubTypes;
import com.fasterxml.jackson.annotation.JsonTypeInfo;
import io.fabric8.kubernetes.client.utils.Serialization;

public class Root {

    @JsonTypeInfo(
            use = JsonTypeInfo.Id.NAME,
            include = JsonTypeInfo.As.EXISTING_PROPERTY,
            property = "type"
    )
    @JsonSubTypes(
        @JsonSubTypes.Type(value = Typed.class, name = "x")
    )
    public interface Typeable {

        String getType();

    }

    public static class Typed implements Typeable {

        @Override
        public String getType() {
            return "x";
        }

    }

    private Typeable typeable;

    public Typeable getTypeable() {
        return typeable;
    }

    public void setTypeable(Typeable typeable) {
        this.typeable = typeable;
    }

    public static void main(String[] args) {
        Root root = new Root();
        root.setTypeable(new Typed());
        System.out.println(Serialization.asYaml(root));
    }

}
```

prints

```
---
typeable: !<x>
  type: "x"
```

This is covered in https://stackoverflow.com/questions/40272165/jackson-yaml-type-info-is-wrong-on-serialization and seems to require disabling a feature on the YAMLFactory.

**Comment by manusa** [comment]
Was this issue fixed by #2990 ? can we close it?

**Comment by shawkins** [comment]
Yes it was fixed and can be closed.
